MxAnyType/include/AnyType.h

```diff
@@ -0,0 +1,65 @@
+#ifndef ANYTYPE_H
+#define ANYTYPE_H
+
+#include <Eigen/Core>
```

Comment:
I don't see any problem at all if `AnyType.h` is not part of a public library, i.e. a C++ library that we will expect to use outside the WB-Toolbox repository. Even if `AnyType.h` will eventually be exported as a C++ library, including the Eigen headers in the public headers is not a impossible, but it creates the following problems: 
* it complicates writing the `PackageConfig.cmake` file,
* it can create problems if there are several Eigen version in the system/bazel workspace/cmake superbuild (unfortunately some projects vendor their own non-system Eigen version, such as TensorFlow or Drake),
* it can create ABI incompatibilities if downstream user/libraries change the Eigen ABI using the Eigen compile-time options (see https://github.com/flexible-collision-library/fcl/issues/96#issuecomment-205229439).

Reply:
Thanks @traversaro! Let's keep things simple and let's use only `std::vector` for returning data. The only issue I see here is efficiency, the raw array created by Matlab is copied once.

Actually, it is copied twice, because mex functions already perform a deep copy of the internal representation used by Matlab. There are however undocumented functions which allow avoiding this copy, and I'd like to check them sooner or later. It would be nice having in the interface also the support of 0-copy methods (for sure we would have to return C-style arrays), handy when data is big. This is not our case, and this will be a future activity.

Reply:
If this code is only used at configuration time, I would worry for efficiency only if it emerges as a problem in actual usage. Having said that, I think the cleanest way to support this kind of pattern is to implement a simple structure;
~~~cpp
struct ExternalVector
{
    ExternalVector() {}
    ExternalVector(double * in_data, size_t in_size): data(in_data), size(in_size) {} 
    double * data{nullptr};
    size_t size{0};
}
~~~
and add a `asVectorDouble(const ExternalVector & data)` method.  
In this way the user-code can provide a simple function: 
~~~cpp
inline ExternalData fromEigen(Eigen::VectorXd& vec) { return ExternalData(vec.data(), vec.size()); }  
~~~
And happy use the method without any additional copy.